spec.md

```diff
@@ -190,18 +190,28 @@ within the same JSON object.
 ### source
 
 - Type: `URI-reference`
-- Description: This describes the event producer. Often this will include
-  information such as the type of the event source, the organization publishing
-  the event, the process that produced the event, and some unique identifiers.
-  The exact syntax and semantics behind the data encoded in the URI is event
-  producer defined.
+- Description: Identifies the event producer. Often this will include
+  information such as the type of the event source, the organization
+  publishing the event, the process that produced the event, and some unique
+  identifiers. The exact syntax and semantics behind the data encoded in the URI
+  is event producer defined.
+
+  An application MUST assign a distinct `source` to each distinct producer.
+  The application MAY use UUIDs, URNs, DNS authorities or an application-specific
+  scheme to create unique identifiers.
+
 - Constraints:
   - REQUIRED
 - Examples
-  - https://github.com/cloudevents/spec/pull
-  - /cloudevents/spec/pull
-  - urn:event:from:myapi/resource
-  - mailto:cncf-wg-serverless@lists.cncf.io
+  - Internet-wide unique URI with a DNS authority.
+    - https://github.com/cloudevents/spec/pull/123
```

Comment:
Suggest that the example here uses `https://github.com/cloudevents/spec` (or some parent path), and move the rest of the URL into the `subject` property.

This makes it much easier for users to write meaningful source-based filters without needing to resort to wildcards.

Reply:
Fixed in next update